upstream-community-operators/instana-agent/instana-agent.v0.0.1.clusterserviceversion.yaml

```diff
@@ -0,0 +1,143 @@
+     mediatype: image/svg+xml
+  version: 0.0.1
+  minKubeVersion: 1.11.0
+  provider:
+    name: Instana
+  links:
+    - name: GitHub Repository
+      url: https://github.com/instana/instana-agent-operator
+  keywords: ['monitoring', 'APM', 'Instana']
+  maintainers:
+    - email: support@instana.com
+      name: Instana
+  maturity: alpha
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+      - description: Instana Agent Leader
+        displayName: Instana Agent Leader
+        kind: ElectedLeader
+        name: electedleaders.agent.instana.io
+        statusDescriptors:
+          - description: Pod name of the current Instana Agent Leader
+            displayName: Elected Leader
+            path: leaderName
+        version: v1alpha1
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+        - serviceAccountName: instana-agent-operator
+          rules:
+          # TODO: This is an experimental alpha version. We didn't clean up the permissions yet,
+          # so these are probably more rights than needed for running the operator.
+          # This will be cleaned up before promoting the operator to beta.
+          - apiGroups: ["", "apps", "extensions", "rbac.authorization.k8s.io"] # "" indicates the core API group
+            resources:
+            - pods
+            - pods/exec
+            - services
+            - endpoints
+            - persistentvolumeclaims
+            - events
+            - configmaps
+            - secrets
+            - replicasets
+            - deployments
+            - daemonsets
+            - clusterroles
+            - clusterrolebindings
+            - serviceaccounts
+            verbs:
+            - '*'
+          - apiGroups: ["apiextensions.k8s.io", "customresourcedefinitions.apiextensions.k8s.io"]
+            resources:
+            - customresourcedefinitions
+            verbs:
+            - list
+          - apiGroups: ["agent.instana.io"]
+            resources:
+            - electedleaders
+            verbs:
+            - '*'
+      deployments:
+        - name: instana-agent-operator
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                k8s-app: instana-agent-operator
+            template:
+              metadata:
+                labels:
+                  k8s-app: instana-agent-operator
+              spec:
+                serviceAccount: instana-agent-operator
+                containers:
+                - image: instana/instana-agent-operator:0.0.1
+                  name: instana-agent-operator
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: POD_NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.namespace
```

Comment:
Seems that this operator was not written with the SDK. How should the operator be configured to watch CRs in one namespace versus all namespaces (clusterwide)?
Usually, we require operators use the OLM added annotation at `metadata.annotations['olm.targetNamespaces']` which is set to the namespace(s) that the operator should watch. In the AllNamespaces case, the annotation is set to the empty string to tell the operator to watch the whole cluster. Does this operator support this configuration?

Reply:
We didn't implement `metadata.annotations['olm.targetNamespaces']`, but we can add it. The operator will just watch the namespace where it is running in, which is `instana-agent` in most cases.

We didn't find the time to implement proper custom resources for everything yet, so the operator at this stage requires two config maps to be present.

Here's documentation which resources must be created before creating the subscription for the operator: https://github.com/instana/instana-agent-operator/blob/master/docs/install-via-olm.md